File: Artifact 1/Original/Artifact 1/Shadow-Wizard-Programming-main/UserData.cpp
//Jevonte Frederick
//4/5/25

#include "UserData.h"

using namespace std;

void UserData::setStartingAmount(double t_startAmount) { //Set initial amount
	m_startAmount = t_startAmount;
}
void UserData::setMonthlyDeposit(double t_depositAmount) { //Set monthly deposit
	m_depositAmount = t_depositAmount;
}
void UserData::setInterestRate(double t_interest) { //Set interest rate
	m_interest = (t_interest / 100);
}
void UserData::setYears(int t_years) { //Set years
	m_years = t_years;
}


int UserData::getYears() { //Return years for loop statments
	return m_years;
}

void UserData::setInitialTotalBalance() { //Initialize total balance
	m_totalBalance = m_startAmount;
}
void UserData::setTotalBalance() { //Calculate total balance without monthly deposits
	m_totalBalance = m_totalBalance + m_interestMonthly; //Add monthly interest to total balance 
}
void UserData::setTotalBalanceMonthly() { //Calculate total balance with monthly deposits
	m_totalBalance = m_totalBalance + m_depositAmount; //Add monthly desposit to total balance
}
void UserData::setInitialInterestTotal() { //Initialize interest total to 0
	m_interestTotal = 0.0;
}
void UserData::setInterestTotal() { //Calculate total interest
	m_interestTotal = m_interestTotal + m_interestMonthly; //Add monthly interest to interest total
}
void UserData::setInterestMonthly() { //Calculate monthly interest
	m_interestMonthly = m_totalBalance * (m_interest / 12.0); //Calculate interest for current month
}

double UserData::getTotalBalance() { //Return total balance
	return m_totalBalance;
}
double UserData::getInterestTotal() { //Return total interest
	return m_interestTotal;
}

File: Artifact 1/Enhancement/Artifact 1 Enhancement/interestCalc/src/interestCalc/UserData.java
//By: Jevonte Frederick
//5/23/26

package interestCalc;

public class UserData {
	//User input
	private double startAmount = 0; //Starting amount
	private double depositAmount = 0; //Monthly deposit amount
	private double interest = 0; //Interest rate
	private int years = 0; //Total years for interest report

	//Calculations from user input for report
	private double totalBalance = 0; //Total balance
	private double interestTotal = 0; //Total interest
	private double interestMonthly = 0; //Monthly interest
	
	//Setter methods
	public void setStartingAmount(double startAmount) { //Set initial amount
		this.startAmount = startAmount;
	}
	public void setMonthlyDeposit(double depositAmount) { //Set monthly deposit
		this.depositAmount = depositAmount;
	}
	public void setInterestRate(double interest) { //Set interest rate
		this.interest = (interest / 100);
	}
	public void setYears(int years) { //Set years
		this.years = years;
	}
	public void setInitialTotalBalance() { //Initialize total balance
		totalBalance = startAmount;
	}
	public void setInitialInterestTotal() { //Initialize interest total to 0
		interestTotal = 0.0;
	}

	//Getter methods
	public int getYears() { //Return years
		return years;
	}
	public double getTotalBalance() { //Return total balance
		return totalBalance;
	}
	public double getInterestTotal() { //Return total interest
		return interestTotal;
	}
	
	//Calculation methods
	public void calculateTotalBalance() { //Calculate total balance without monthly deposits
		totalBalance = totalBalance + interestMonthly; //Add monthly interest to total balance 
	}
	public void calculateTotalBalanceMonthlyDeposit() { //Calculate total balance with monthly deposits
		totalBalance = totalBalance + depositAmount; //Add monthly deposit to total balance
	}
	public void calculateInterestTotal() { //Calculate total interest
		interestTotal = interestTotal + interestMonthly; //Add monthly interest to interest total
	}
	public void calculateInterestMonthly() { //Calculate monthly interest
		interestMonthly = totalBalance * (interest / 12.0); //Calculate interest for current month
	}

}
